src/event/value.rs

```diff
@@ -142,7 +143,66 @@ impl From<serde_json::Value> for Value {
     }
 }
 
+impl TryInto<serde_json::Value> for Value {
+    type Error = crate::Error;
+
+    fn try_into(self) -> Result<serde_json::Value, Self::Error> {
+        match self {
+            Value::Boolean(v) => Ok(serde_json::Value::from(v)),
+            Value::Integer(v) => Ok(serde_json::Value::from(v)),
+            Value::Float(v) => Ok(serde_json::Value::from(v)),
+            Value::Bytes(v) => Ok(serde_json::Value::from(String::from_utf8(v.to_vec())?)),
+            Value::Map(v) => Ok(serde_json::to_value(v)?),
+            Value::Array(v) => Ok(serde_json::to_value(v)?),
+            Value::Null => Ok(serde_json::Value::Null),
+            Value::Timestamp(v) => Ok(serde_json::Value::from(timestamp_to_string(&v))),
+        }
+    }
+}
+
 impl Value {
+    pub fn is_boolean(&self) -> bool {
```

Comment:
We're already using the `is_enum_variant` derive for this elsewhere, so could probably use it here too.